rocksdb_admin/admin_handler.cpp

```diff
@@ -929,6 +974,177 @@ void AdminHandler::async_tm_addS3SstFilesToDB(
   callback->result(AddS3SstFilesToDBResponse());
 }
 
+int AdminHandler::getPartitionIDFromDBName(const std::string& db_name) {
+  const size_t n = db_name.find_first_of("0123456789");
+  if (n != std::string::npos){
+    return std::stoi(db_name.substr(n, std::string::npos));
+  }
+  LOG(ERROR) << "No partition id found in db name " << db_name;
+
+  return -1;
+}
+
+std::string AdminHandler::getHostName() {
+  char buffer[512];
+  gethostname(buffer, sizeof(buffer));
+  return std::string(buffer);
+}
+
+std::string AdminHandler::getConsumerGroupId(
+    const std::string& topic_name, const std::string& partition_id) {
+  return getHostName() + '_' + topic_name + '_' + partition_id;
+}
+
+void AdminHandler::async_tm_startMessageIngestion(
+    std::unique_ptr<apache::thrift::HandlerCallback<std::unique_ptr<
+      StartMessageIngestionResponse>>> callback,
+    std::unique_ptr<StartMessageIngestionRequest> request) {
+
+  admin::AdminException e;
+  e.errorCode = AdminErrorCode::DB_ADMIN_ERROR;
+
+  const auto db_name = request->db_name;
+  const auto topic_name = request->topic_name;
+  const auto kafka_broker_serverset_path = request->kafka_broker_serverset_path;
+  const auto replay_timestamp_ms = request->replay_timestamp_ms;
+
+  LOG(INFO) << "Called startMessageIngestion for " << db_name;
+
+  // Check if there's already a thread consuming the same partition.
+  if (sync_var_map.find(db_name) != sync_var_map.end()) {
```

Comment:
access to sync_var_map needs to be synchronized.

Reply:
+1